bindings/c/test/fdb_c_client_config_tests.py

```diff
@@ -338,6 +342,16 @@ def setUpClass(cls):
     def tearDownClass(cls):
         cls.cluster.tear_down()
 
+    def test_tls_disable_plaintext_connection(self):
+        # Local client only
+        test = ClientConfigTest(self)
+        test.print_status = True
+        test.tls_disable_plaintext_connection = True
+        test.transaction_timeout = 100
```

Comment:
Are we expected to fail with timeouts on TLS configuration errors? @sfc-gh-jshim do you know?

Reply:
Typically a failed [connection attempt](https://github.com/apple/foundationdb/blob/main/fdbrpc/FlowTransport.actor.cpp#L702) would result in [retry with backoff](https://github.com/apple/foundationdb/blob/a7cf82adb2294f2468cb79c4b5962da040f89f59/fdbrpc/FlowTransport.actor.cpp#L687).
But now that you mention it, we could do better here.
Whether an endpoint is TLS is known in advance to connection attempt: the `:tls` address suffix.
Instead of re-using `connection_failed`, suffering the wait until timeout, and giving the false impression of some server or infrastructure failure, we could let it fail early on with a different error code: e.g. `plaintext_connection_prevented`. Gatekeeping function could be `TransportData::getOrOpenPeer()`, comparing the destination address' `:tls` suffix with the global network option, only if `FlowTransport::isClient()`.

Reply:
If the TLS errors are retriable, then it is in line with what we have now. We can plan a separate task for improving error reporting on such errors